internal/cli/login.go

```diff
@@ -12,24 +12,68 @@ import (
 	"github.com/auth0/auth0-cli/internal/prompt"
 )
 
+var (
+	loginAsMachine = Flag{
+		Name:       "Login as machine",
+		LongForm:   "as-machine",
+		Help:       "Initiates authentication as a machine via client credentials (client ID, client secret)",
+		IsRequired: false,
+	}
+	loginTenantDomain = Flag{
+		Name:       "Tenant Domain",
+		LongForm:   "domain",
+		Help:       "Specifies tenant domain when authenticating via client credentials (client ID, client secret)",
+		IsRequired: false,
+	}
+)
+
+type LoginInputs struct {
+	LoginAsMachine bool
+	Domain         string
+	ClientID       string
+	ClientSecret   string
+}
+
 func loginCmd(cli *cli) *cobra.Command {
+	var inputs LoginInputs
+
 	cmd := &cobra.Command{
 		Use:   "login",
 		Args:  cobra.NoArgs,
 		Short: "Authenticate the Auth0 CLI",
-		Long:  "Sign in to your Auth0 account and authorize the CLI to access the Management API.",
+		Long:  "Authenticates the Auth0 CLI either as a user using personal credentials or as a machine using client credentials (client ID/secret).",
+		Example: `
+		auth0 login
+		auth0 login --as-machine
+		auth0 login --as-machine --domain <TENANT_DOMAIN> --client-id <CLIENT_ID> --client-secret <CLIENT_SECRET>
+		`,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
-			if _, err := RunLogin(ctx, cli, false); err != nil {
-				return err
+
+			shouldLoginAsMachine := inputs.LoginAsMachine || inputs.ClientID != "" || inputs.ClientSecret != "" || inputs.Domain != ""
+
+			if shouldLoginAsMachine {
+				if err := RunLoginAsMachine(ctx, inputs, cli, cmd); err != nil {
+					return err
+				}
+			} else {
+				if _, err := RunLoginAsUser(ctx, cli, false); err != nil {
+					return err
+				}
 			}
 
+			cli.renderer.Infof("Successfully authenticated to %s", inputs.Domain)
 			cli.tracker.TrackCommandRun(cmd, cli.config.InstallID)
 
 			return nil
 		},
 	}
 
+	loginAsMachine.RegisterBool(cmd, &inputs.LoginAsMachine, false)
+	loginTenantDomain.RegisterString(cmd, &inputs.Domain, "")
+	tenantClientID.RegisterString(cmd, &inputs.ClientID, "")
```

Comment:
Since `tenants add` is being removed, `tenantClientID` and `tenantClientSecret` are no longer being used in the `tenants.go` file. Should we move their definitions to this file?